qiskit/circuit/library/arithmetic/adders/classicaladder.py

```diff
@@ -0,0 +1,119 @@
+
+from qiskit.circuit import QuantumCircuit, QuantumRegister, AncillaRegister
+
+
+class ClassicalAdder(QuantumCircuit):
+    r"""A ClassicalAdd Circuit to perform in-place addition on two qubit registers.
+     Circuit to compute the sum of two qubit registers using the Classical Addition Part from [1].
+
+     Given two equally sized input registers that store quantum states
+    :math:`|a\rangle` and :math:`|b\rangle`, performs addition of numbers that
+    can be represented by the states, storing the resulting state in-place in the second register:
+    .. math::
+        |a\rangle |b\rangle \mapsto |a\rangle |a+b\rangle
+    Here :math:`|a\rangle` (and correspondingly :math:`|b\rangle`) stands for the direct product
+    :math:`|a_n\rangle \otimes |a_{n-1}\rangle \ldots |a_{1}\rangle \otimes |a_{0}\rangle`
+    which denotes a quantum register prepared with the value :math:`a = 2^{0}a_{0} + 2^{1}a_{1} +
+    \ldots 2^{n}a_{n}` [2].
+    As an example, a classical add circuit that performs addition on two 3-qubit sized
+    registers is as follows:
+    .. parsed-literal::
+           ┌────────┐                                                   ┌────────┐┌──────┐
+input_a_0: ┤1       ├───────────────────────────────────────────────────┤1       ├┤1     ├
+           │        │┌────────┐                       ┌────────┐┌──────┐│        ││      │
+input_a_1: ┤        ├┤1       ├───────────────────────┤1       ├┤1     ├┤        ├┤      ├
+           │        ││        │┌────────┐     ┌──────┐│        ││      ││        ││      │
+input_a_2: ┤        ├┤        ├┤1       ├──■──┤1     ├┤        ├┤      ├┤        ├┤      ├
+           │        ││        ││        │  │  │      ││        ││      ││        ││      │
+input_b_0: ┤2       ├┤        ├┤        ├──┼──┤      ├┤        ├┤      ├┤2       ├┤2     ├
+           │        ││        ││        │  │  │      ││        ││      ││        ││  Sum │
+input_b_1: ┤  Carry ├┤2       ├┤        ├──┼──┤      ├┤2       ├┤2     ├┤  Carry ├┤      ├
+           │        ││        ││        │┌─┴─┐│      ││        ││  Sum ││        ││      │
+input_b_2: ┤        ├┤  Carry ├┤2       ├┤ X ├┤2     ├┤  Carry ├┤      ├┤        ├┤      ├
+           │        ││        ││  Carry │└───┘│  Sum ││        ││      ││        ││      │
+   cout_0: ┤        ├┤        ├┤3       ├─────┤      ├┤        ├┤      ├┤        ├┤      ├
+           │        ││        ││        │     │      ││        ││      ││        ││      │
+    cin_0: ┤0       ├┤        ├┤        ├─────┤      ├┤        ├┤      ├┤0       ├┤0     ├
+           │        ││        ││        │     │      ││        ││      ││        │└──────┘
+    cin_1: ┤3       ├┤0       ├┤        ├─────┤      ├┤0       ├┤0     ├┤3       ├────────
+           └────────┘│        ││        │     │      ││        │└──────┘└────────┘
+    cin_2: ──────────┤3       ├┤0       ├─────┤0     ├┤3       ├──────────────────────────
+                     └────────┘└────────┘     └──────┘└────────┘ 
```

Comment:
This needs to be indented with 8 more spaces so that it looks like
```
    """
    text before the image...

    .. parsed-literal:

        input_a_0: ....